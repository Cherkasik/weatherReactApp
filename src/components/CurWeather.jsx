import React from 'react';
import styled from 'styled-components';
import InfoBlock from './InfoBlock';

const Container = styled.div`
  display: flex;
  justify-content: space-between;
`;

const InfoContainer = styled.div`
  width: 45%;
`;

const PlaceName = styled.div`
  font-size: 30pt;
  font-weight: 700;
  color: #71848B;
`;

const Icon = styled.div`
  width: 100px;
  height: 100px;
  float: left;
  border: 3px solid #71848B;
  margin: 40px 0 0 0;
  text-align: center;
  color: #6E8084;
  
  p {
    margin: 31px 0;
  }
`;

const Weather = styled.div`
  float: right;
  font-size: 80pt;
  line-height: 80px;
  color: #869197;
  margin: 40px 40px 0 0;
`;


function CurWeather() {
  return (
    <Container>
      <InfoContainer>
        <PlaceName>Санкт-Петербург</PlaceName>
        <Icon>
          <p>Иконка погоды</p>
        </Icon>
        <Weather>8°С</Weather>
      </InfoContainer>
      <InfoContainer>
        <InfoBlock />
      </InfoContainer>
    </Container>
  );
}

export default CurWeather;
