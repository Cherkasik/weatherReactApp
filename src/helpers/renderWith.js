import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import reducer from '../store/rootReducer';

export function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  }
}
