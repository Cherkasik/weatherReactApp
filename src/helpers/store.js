import thunkMiddleware from "redux-thunk";
import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../store/rootReducer";

const middlewares = [thunkMiddleware];

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger');
  middlewares.push(logger);
}

export const store = compose(applyMiddleware(...middlewares))(createStore)(rootReducer);