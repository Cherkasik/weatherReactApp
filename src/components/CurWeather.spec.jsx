import React from 'react';
import 'jest-styled-components';
import { renderWithRedux as render } from "../helpers/renderWith";
import CurWeather from './CurWeather';

describe('CurWeather component', () => {
  it('should render correctly', () => {
    const { container } = render(<CurWeather />, {
      initialState: {
        weather: {
          weather: {
            name: 'SPb',
            main: { temp: 283, pressure: 10000, humidity: 10 },
            weather: [{ icon: 10 }],
            wind: { speed: 10, deg: 10 },
            clouds: { all: 10 },
            coord: { lon: 10, lat: 10 },
          }
        }
      }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});