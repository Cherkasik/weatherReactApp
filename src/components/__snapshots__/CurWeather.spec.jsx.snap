// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurWeather component should render correctly 1`] = `
.c5 {
  width: 97.5%;
  background: #A7AFB1;
  margin-top: 10px;
  padding: 10px;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 div:first-child {
  font-size: 14pt;
  font-weight: 700;
}

.c5 div:last-child {
  font-weight: 500;
  font-size: 14pt;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 45%;
}

.c2 {
  font-size: 30pt;
  font-weight: 700;
  color: #71848B;
}

.c3 {
  width: 100px;
  height: 100px;
  float: left;
  border: 3px solid #71848B;
  margin: 40px 0 0 0;
  color: #6E8084;
}

.c3 img {
  width: 100px;
  height: 100px;
}

.c4 {
  float: right;
  font-size: 80pt;
  line-height: 80px;
  color: #869197;
  margin: 40px 40px 0 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      SPb
    </div>
    <div
      class="c3"
    >
      <img
        alt="icon"
        src="https://openweathermap.org/img/w/10.png"
      />
    </div>
    <div
      class="c4"
    >
      10
      °C
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c5"
    >
      <div>
        Ветер
      </div>
      <div>
        10
         м/с, 
        10
        °
      </div>
    </div>
    <div
      class="c5"
    >
      <div>
        Облачность
      </div>
      <div>
        10
        %
      </div>
    </div>
    <div
      class="c5"
    >
      <div>
        Давление
      </div>
      <div>
        10
         Бар
      </div>
    </div>
    <div
      class="c5"
    >
      <div>
        Влажность
      </div>
      <div>
        10
        %
      </div>
    </div>
    <div
      class="c5"
    >
      <div>
        Координаты
      </div>
      <div>
        [
        10
        , 
        10
        ]
      </div>
    </div>
  </div>
</div>
`;
